Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM node:latest
+
+ENV UID=991 GID=991
+
+RUN mkdir -p /app
+WORKDIR /app
+
+COPY ./ /app
+
+RUN npm install -g typescript @angular/cli \
+      && cd app \
+      && npm install \
+      && ng build \
+      && cd .. \
+      && npm install \
+      && tsc
```

Comment:
is it a best practice to re-build the app vs. just doing `npm i mongoku`?

Reply:
It's a matter of preference, but this way builds the version you have checked out of git, while the your example it pulls in the last release pushed to the npm, which might not be the same version.

I think it would be confusing to make changes to the source code, rebuild the image, and still have the old source code running because it pulled it using npm.

Ideally generating the docker image and publishing to docker hub would be part of CI whenever a "release" is tagged.

Reply:
Got it, and it makes perfect sense. Thanks!